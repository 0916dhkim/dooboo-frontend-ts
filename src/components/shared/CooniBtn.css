.authBtn {
  width: 320px;
  height: 50px;
  border-radius: 3px;
  border-style: solid;
  border-width: 2px;
  margin: 10px 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.2s ease-in-out;

  &:hover {
    opacity: 0.5;
  }
  &:active {
    opacity: 0.5;
  }

  @media (max-width: 740px) {
    width: 85vw;
    height: 8vh;
  }
}

.logoImg {
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover;
}