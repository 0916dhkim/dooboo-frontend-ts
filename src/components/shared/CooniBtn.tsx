import React, { Component } from 'react';
import Radium from 'radium';

import { effects } from '@utils/styles';

export class CooniBtn extends Component<any> {
  public render() {
    return (
      <div
        className='hi'
        style={[
          styles.authBtn,
          this.props.white
            ? {borderColor: '#fff', color: '#fff'}
            : {borderColor: '#000', color: '#000'},
        ]}
        onClick={() => this.props.onClick()}
      >
        {
          this.props.imgSrc
            ? <img
                src={this.props.imgSrc}
                srcSet={this.props.srcset}
                style={styles.logoImg}
              />
            : null
        }
        <span style={styles.authTxt}>{this.props.btnTxt}</span>
      </div>
    );
  }
}

const styles: any = {
  authBtn: {
    width: '320px',
    height: '28px',
    borderRadius: '3px',
    borderStyle: 'solid',
    borderWidth: '2px',
    margin: '8px 0',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',

    ...effects.btnEffect,

    ':hover': {
      ...effects.btnEffectHover,
    },
    ':active': {
      opacity: 0.5,
    },

    '@media (max-width: 740px)': {
      width: '85vw',
      height: '8vh',
    },
  },

  logoImg: {
    position: 'absolute',
    left: '16px',
    height: '20px',
    width: '20px',
    objectFit: 'cover',
  },
};

export default Radium(CooniBtn);
