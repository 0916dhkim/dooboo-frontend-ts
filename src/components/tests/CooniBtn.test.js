import React from 'react';
import renderer from 'react-test-renderer';

import CooniBtn from '@shared/CooniBtn';

test('CooniBtn changes border color when it is clicked', () => {
  const cooniBtn = renderer.create(
    <CooniBtn white={true} clickHandler={props => props.white = !props.white} btnTxt='jest is awesome' />
  );
  let tree = cooniBtn.toJSON();
  expect(tree).toMatchSnapshot();
  console.log(tree);
  console.log(tree.props);
//   console.log(tree.children);
//   console.log(tree.children[0]);
//   console.log(tree.children[0].props.onClick);
//   console.log(tree.children[0].props.onClick[0]);

//   tree.props.clickHandler();
//   tree = cooniBtn.toJSON();
//   expect(tree).toMatchSnapshot();
});