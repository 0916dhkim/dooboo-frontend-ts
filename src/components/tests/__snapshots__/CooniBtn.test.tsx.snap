// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CooniBtn shared component test component and snapshot matches 1`] = `
<div
  className="hi"
  onClick={[Function]}
  style={
    Array [
      Object {
        ":active": Object {
          "opacity": 0.5,
        },
        ":hover": Object {
          "opacity": 0.5,
        },
        "@media (max-width: 740px)": Object {
          "height": "8vh",
          "width": "85vw",
        },
        "alignItems": "center",
        "borderRadius": "3px",
        "borderStyle": "solid",
        "borderWidth": "2px",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "height": "50px",
        "justifyContent": "center",
        "margin": "10px 0",
        "opacity": 1,
        "transition": "opacity 0.2s ease-in-out",
        "width": "320px",
      },
      Object {
        "borderColor": "#fff",
        "color": "#fff",
      },
    ]
  }
>
  <span>
    CooniBtn 2nd test
  </span>
</div>
`;

exports[`CooniBtn shared component test component onClick function has to be added and work to make the count 2 1`] = `
<div
  className="hi"
  onClick={[Function]}
  style={
    Array [
      Object {
        ":active": Object {
          "opacity": 0.5,
        },
        ":hover": Object {
          "opacity": 0.5,
        },
        "@media (max-width: 740px)": Object {
          "height": "8vh",
          "width": "85vw",
        },
        "alignItems": "center",
        "borderRadius": "3px",
        "borderStyle": "solid",
        "borderWidth": "2px",
        "cursor": "pointer",
        "display": "flex",
        "flexDirection": "row",
        "height": "50px",
        "justifyContent": "center",
        "margin": "10px 0",
        "opacity": 1,
        "transition": "opacity 0.2s ease-in-out",
        "width": "320px",
      },
      Object {
        "borderColor": "#fff",
        "color": "#fff",
      },
    ]
  }
>
  <span>
    CooniBtn 2nd test
  </span>
</div>
`;
